cfg/agentinfo/info.go

```diff
@@ -72,6 +72,10 @@ func UserAgent() string {
 	return userAgent
 }
 
+func ResetUserAgent() {
```

Comment:
This was added to reset the user agent during the tests. Should add a comment so that others can know that it's not meant to be used normally.

Reply:
Thinking through this a little more in the concurrency model, is there a possibility of unsetting this as it's being passed to the handler?

Reply:
As discussed, this has been removed. Also added a map to store the user agent string for each log group.